src/prefect/cli/deployment.py

```diff
@@ -473,45 +466,65 @@ async def build(
         )
     else:
         # default storage, no need to move anything around
-        storage = LocalFileSystem(basepath=Path(".").absolute())
+        storage = None
+        deployment_path = str(Path(".").absolute())
 
     # persists storage now in case it contains secret values
-    await storage._save(is_anonymous=True)
+    if storage and not storage._block_document_id:
+        await storage._save(is_anonymous=True)
 
     if infra_block:
-        template = await Block.load(infra_block)
-        infrastructure = template.copy(
-            exclude={"_block_document_id", "_block_document_name", "_is_anonymous"}
-        )
-    else:
+        infrastructure = await Block.load(infra_block)
+    elif infra_type:
         if infra_type == Infra.kubernetes:
             infrastructure = KubernetesJob()
         elif infra_type == Infra.docker:
             infrastructure = DockerContainer()
-        else:
+        elif infra_type == Infra.process:
             infrastructure = Process()
+    else:
+        infrastructure = None
```

Comment:
```suggestion
        infrastructure = Process()
```

to support incremental adoption; otherwise leaving --infra or -ib out will result in:
```
  File "/Users/anna/repos/prefect/src/prefect/cli/deployment.py", line 514, in build
    deployment = Deployment(
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Deployment
infrastructure
  none is not an allowed value (type=type_error.none.not_allowed)
```

when doing simple:
```
prefect deployment build flows/healthcheck.py:healthcheck --name dev --tag dev
```